config.yaml

```diff
@@ -1,14 +1,15 @@
 # Copyright 2021 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# TEMPLATE-TODO: change this example to suit your needs.
-# If you don't need a config, you can remove the file entirely.
-# It ties in to the example _on_config_changed handler in src/charm.py
-#
-# Learn more about config at: https://juju.is/docs/sdk/config
 
 options:
-  thing:
-    default: 🎁
-    description: A thing used by the charm.
+  mysql_root_password:
     type: string
+    default: ''
+    descriptions: Password for root user, equivalent to MYSQL_ROOT_PASSWORD.
+  mysql_user:
+    type: string
+    default: ''
+    description: MySQL username for privileged superuser. This is used in conjunction with MYSQL_PASSWORD. Equivalent to MYSQL_USER.
+  mysql_password:
+    type: string
+    description: Password for MYSQL_USER user.
```

Comment:
This works as an intermediary way of configuring it, but once you've gotten a bit further in the implementation, it should probably be generated, and retrieved by the operator using an action.

Reply:
Passwords stored in plain text in a file are considered bad practice. Generating them at deployment time will be safer (the option to query must be provided).  One important consideration is that with every deployment a new password would be generated and this could be a problem for some enterprises.  One step further would be to generate upon first deployment, store in a secure repository (Juju secrets?) and reuse on subsequent deployments.  Yet another consideration should be when a company has a strict password expiry policy.

Reply:
The pattern we use in the O11y team is to have the operator automatically generate the passwords, and provide and action for the administrator to retrieve them

Reply:
The intention is to default these passwords to an empty string and have the user change that through config, when a ConfigChanged is triggered, these defaults will be wiped away.

Reply:
> The intention is to default these passwords to an empty string and have the user change that through config

Thanks for the clarification! 🙏🏽 

What I'm suggesting is that these defaults probably should be generated as well, and that setting them through config values likely isn't something we want in the long-run.

See https://github.com/canonical/grafana-k8s-operator/blob/e3b5795cb3ff0da328549b3e266bfb8c1ebb4047/src/charm.py#L539-L562 for reference. If/Once the user sets them to something else (in this case using for instance the mysql-cli), the action to fetch the default password would be disabled.

Reply:
I absolutely agree with generating random defaults, that's better. Will change that.

Reply:
Thanks for the suggestions. I've removed the `config.yaml` and added a new helper method in the charm code to generate a random password.